components/forms/TaskForm.tsx

```diff
@@ -0,0 +1,136 @@
+import { Checkbox, DatePicker, Form, FormInstance, Input, InputNumber, Select, TimePicker } from 'antd';
+import React, { useEffect, useState } from 'react';
+import { NotificationDateTimeType, TaskFormData } from '../../types/task';
+import { Nullable } from '../../types/utils';
+
+interface Props {
+  initialValues: TaskFormData;
+  shouldTouchAllCompulsoryFields: boolean;
+  shouldShowMarkDone: boolean;
+  isSubmitting: boolean;
+  setShouldDisableSaveButton: React.Dispatch<React.SetStateAction<boolean>>;
+  setTaskFormData: React.Dispatch<React.SetStateAction<Nullable<TaskFormData>>>;
+}
+
+const { Option } = Select;
+
+function TaskForm({
+  initialValues,
+  shouldTouchAllCompulsoryFields,
+  shouldShowMarkDone,
+  isSubmitting,
+  setShouldDisableSaveButton,
+  setTaskFormData,
+}: Props) {
+  const [form] = Form.useForm();
+
+  const [shouldShowNotificationDaysOffsetInput, setShouldShowNotificationDaysOffsetInput] = useState(false);
+  const [shouldShowNotificationDateTimePicker, setShouldShowNotificationDateTimePicker] = useState(false);
+  const [shouldShowNotificationTimePicker, setShouldShowNotificationTimePicker] = useState(false);
+
+  useEffect(() => {
+    form.resetFields();
+    onSelectNotificationDateTimeType(initialValues.notificationDateTimeType ?? NotificationDateTimeType.NONE);
+  }, [initialValues]);
```

Comment:
```suggestion
  }, []);
```

Am I right to say that `initialValues` will never be changed after this component mounts?

Reply:
No.